### DISCUSSION TITLE
Prettyprompt + Parser ?

### DISCUSSION THREAD

**Discussion by SadE54** [waf/PrettyPrompt#220] [Q&A] [answered]
Hi ,

I'm starting to develop  a sort of REPL application for a domain specific language. Pretty prompt seems really cool to use with spectre console for that , but I have one question.
I'm using a parser, This parser is able to call specific functions when specific tokens of the grammar are detected. I want to use this possibility to change dynamically the list for the autocompletion according the prompt content state . 
Is there currently a callback for each key press , to parse what was entered , and change the autocompletion list in PrettyPrompt ? 

Regards,

Yann

**Comment by waf** [comment, marked as answer]
Hey @SadE54, sorry I missed this. We don't currently have any support for changing the autocompletion list dynamically.

Would you prefer complete manual control (e.g. manual calls to close/open/set suggestions) or have something else in mind? I can't promise it would get implemented but I'm open to adding additional methods to the API surface.

**Reply by SadE54** [reply]
Hi, no problem. What you re suggesting would be perfect. 👍
